app/controllers/prepay_balance_controller.js

```diff
@@ -0,0 +1,1557 @@
+//  === Requirements ===
+const { query } = require('express')
+const faker = require('faker')
+
+//  === Controller ===
+module.exports = {
+    index_product_buckets: function(request) 
+    {
+        console.log("prepay_balance_controller.index_product_buckets")
+
+        return [
+            {
+                id: faker.random.uuid(),
+                routeDetails: {
+                    controller: 'Prepay Balance',
+                    action: 'index_product_buckets'
+                },
+                href: faker.internet.url(),
+                name: faker.name.findName(),
+                description: faker.lorem.sentence(),
+                bucketType: "promotionalvoice" ,
+                remainedAmount: [
+                    {
+                        amount: faker.random.number(),
+                        units: "EUR"
+                    }
+                ],
+                reservedAmount:[
+                    {
+                        amount: faker.random.number(),
+                        units: "EUR"
+                    }
+                ],
+                validFor: [
+                    {
+                        startDateTime: new Date(),
+                        endDateTime: new Date()
+                    }
+            
+                ],
+                status: "active",
+                product: [
+                    {
+                        id: faker.random.uuid(),
+                        href: faker.internet.url(),
+                        name: faker.name.findName()
+                    }
+                ],
+                partyAccount: [
+                    {
+                        id: faker.random.uuid(),
+                        href: faker.internet.url(),
+                        name: faker.name.findName()
+                    }
+                ],
+                realizingResource: [
+                    {
+                        id: faker.random.uuid(),
+                        href: faker.internet.url(),
+                        name: faker.name.findName(),
+                        value: "Resourse value"
+                    }
+                ],
+                relatedParty: [
+                    {
+                        id: faker.random.uuid(),
+                        href: faker.internet.url(),
+                        name: faker.name.findName(),
+                        role: "Customer"
+                    }
+                ]
+            }
+        ]
+    },
+    show_product_buckets: function(request)
+    {
+        console.log("prepay_balance_controller.show_product_buckets")
+
+        return {
+            id: faker.random.uuid(),
```

Comment:
please use the id being passed with the request when available (show, update, and deletes)

Reply:
send back the bucket id